pkg/api/norman/customization/cluster/validator.go

```diff
@@ -432,13 +432,16 @@ func validateAKSNodePools(spec *v32.ClusterSpec) error {
 		return nil
 	}
 	if len(nodePools) == 0 {
-		return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("must have at least one nodepool"))
+		return httperror.NewAPIError(httperror.InvalidBodyContent, "must have at least one nodepool")
 	}
 
 	for _, np := range nodePools {
 		name := np.Name
 		if to.String(name) == "" {
-			return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("nodePool Name cannot be an empty string"))
+			return httperror.NewAPIError(httperror.InvalidBodyContent, "nodePool Name cannot be an empty string")
+		}
+		if np.OsType == "Windows" {
+			return httperror.NewAPIError(httperror.InvalidBodyContent, "windows node pools are not supported")
```

Comment:
This means it will take effect on the API layer, but you could technically make one with a plain cluster CR and you could also register one. Registration case I hadn't considered yet. We could either error or send warning.

Reply:
If you made a plain cluster CR you would run into this https://github.com/rancher/aks-operator/pull/17/files#diff-1aa54d4941ca80708078867e1b130aff086719689479dea38683fded2f3c7852R391-R392

I'm not sure what to do about registration.

Reply:
I see two potential options:
1. Release note it saying it is possible to import a cluster with Windows nodes, but it is not supported.
2. We could do some import validation of the AKS cluster in `aks-operator`. We don't do this for imported cluster in any of the the three operators, but we could do it in this case if it is deemed important enough.

Reply:
> Release note it saying it is possible to import a cluster with Windows nodes, but it is not supported.

Would that work though if all these fields are deleted ? The operator/rancher code would treat it like Linux nodes right ? And then who knows what happens. I think we should reject it.

Reply:
The windowsProfile fields would be ignored when the cluster is imported, and they wouldn't be manageable from rancher. But there's nothing in the operator code that would treat the node pools differently. It's just that the monitoring chart still won't work on them.

Reply:
Interesting. So we go that route and then QA it to see if we can break it ? Do linux nodes in aks get the default linux node label by default ? https://github.com/rancher/charts/blob/dev-v2.6/charts/rancher-logging/rancher-logging/3.9.401%2Bup3.9.4/values.yaml#L40-L41

Reply:
Yes, linux nodes get the `kubernetes.io/os=linux` label and windows nodes get `kubernetes.io/os=windows` in AKS. AKS does not allow creating a cluster with windows nodes without also having linux nodes, so the logging chart will always have some linux nodes available where that selector will work.

Reply:
K, lets give that a shot and then release note it.

Reply:
I would imagine we could check the upstreamSpec on status and add validation to a controller right?

Reply:
upstreamSpec status doesn't get set until after the cluster object is created so I don't think that validation would be doable here in the norman pre-create validator, we'd have to put it somewhere else where it wouldn't be useful to the UI.